shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -69,28 +84,38 @@
     }
   }
 
-  [view.surfaceManager present:surfaces
-                        notify:^{
-                          PresentPlatformViews(view, layers, layers_count);
-                        }];
+  CFTimeInterval presentation_time = 0;
+
+  if (layers_count > 0 && layers[0]->presentation_time != 0) {
+    presentation_time = [time_converter_ engineTimeToCAMediaTime:layers[0]->presentation_time];
+  }
+
+  // Notify block below may be called asynchronously, hence the need to copy
+  // the layer information instead of passing the original pointers from embedder.
```

Comment:
Thanks for adding this comment to explain why we couldn't for example, just use a `std::span` or otherwise to wrap the layers being passed down, which will potentially be collected before they're used by us.